libp2p/stream/bufferstream.nim

```diff
@@ -107,10 +109,14 @@ method readOnce*(s: BufferStream,
 
   if rbytes < nbytes:
     # There's space in the buffer - consume some data from the read queue
-    trace "popping readQueue", s, rbytes, nbytes
-    let buf = await s.readQueue.popFirst()
-
-    if buf.len == 0 or s.isEof: # Another task might have set EOF!
+    s.reading = true
```

Comment:
I tried adding this but it didn't seem to ever trigger, so I left it out for now.

Reply:
btw, this should be moved to `LPStream`.

Reply:
it didn't trigger because typically it would deadlock before, in the `reset` loop. moving it to lpstream doesn't quite make sense since each stream handles this a bit differently, ie chronosstream handles the single-reader requirement "internally" already. what we really need is to expose `reset` to nbc as well because that's what it should do when its timeout is reached - notably, the nbc timeout is different than the libp2p - libp2p restarts timer on _any_ activity while nbc counts complete chunk time